--- utvideo-15.3.0/utv_logl/LogReader.cpp.orig	2015-11-24 20:23:07.295710349 +0100
+++ utvideo-15.3.0/utv_logl/LogReader.cpp	2015-11-24 20:23:10.349043583 +0100
@@ -222,7 +222,9 @@
 	pthread_mutex_init(&mutexLock, NULL);
 
 	memset(&sun, 0, sizeof(sun));
+#if defined(__APPLE__)
 	sun.sun_len = sizeof(sun);
+#endif
 	sun.sun_family = AF_UNIX;
 	if (GetLogSocketPath(sun.sun_path, _countof(sun.sun_path)) != 0)
 		return -1;
--- utvideo-15.3.0/utv_logl/LogWriter.cpp.orig	2015-09-22 21:58:24.000000000 +0200
+++ utvideo-15.3.0/utv_logl/LogWriter.cpp	2015-11-24 20:38:38.805671266 +0100
@@ -1,6 +1,7 @@
 /* ï∂éöÉRÅ[ÉhÇÕÇrÇiÇhÇr â¸çsÉRÅ[ÉhÇÕÇbÇqÇkÇe */
 /* $Id$ */
 
+#include <errno.h>
 #include "stdafx.h"
 #include "LogPath.h"
 #include "LogWriter.h"
@@ -116,7 +117,11 @@
 		return -1;
 	}
 
+#if defined(__GLIBC__)
+	sprintf(buf, "%s[%d]\n", program_invocation_short_name, getpid());
+#else
 	sprintf(buf, "%s[%d]\n", getprogname(), getpid()); // XXX getprogname: only BSD family and MacOS
+#endif
 	if (writeLog(buf, strlen(buf)) != strlen(buf))
 	{
 		close(fdLogSock);
--- utvideo-18.2.1/utv_core/HuffmanCode_x86_HuffmanDecode.cpp.orig	2017-07-16 08:59:00.000000000 +0200
+++ utvideo-18.2.1/utv_core/HuffmanCode_x86_HuffmanDecode.cpp	2017-07-22 20:37:15.284141254 +0200
@@ -125,6 +125,7 @@
 .endm
 	HUFFMAN_DECODE_8 )" ARGS R"(
 .purgem HUFFMAN_DECODE_8
+	.att_syntax
 	)"
 		: "=a"(ret), "=D"(clobber), "=S"(clobber), "=b"(clobber)
 		: "D"(pDstBegin), "a"(pDstEnd), "S"(pSrcBegin), "b"(pDecodeTable),
--- utvideo-18.2.1/utv_core/HuffmanCode_x86_HuffmanEncode.cpp.orig	2017-07-16 08:59:00.000000000 +0200
+++ utvideo-18.2.1/utv_core/HuffmanCode_x86_HuffmanEncode.cpp	2017-07-22 20:37:28.217474441 +0200
@@ -92,6 +92,7 @@
 .endm
 	HUFFMAN_ENCODE )" ARGS R"(
 .purgem HUFFMAN_ENCODE
+	.att_syntax
 	)"
 		: "=a"(ret), "=D"(clobber), "=S"(clobber), "=d"(clobber)
 		: "D"(pDstBegin), "S"(pSrcBegin), "a"(pSrcEnd), "d"(pEncodeTable)
--- utvideo-18.2.1/utv_core/HuffmanCode_x64_HuffmanDecode.cpp.orig	2017-07-16 08:59:00.000000000 +0200
+++ utvideo-18.2.1/utv_core/HuffmanCode_x64_HuffmanDecode.cpp	2017-07-22 20:38:44.187473574 +0200
@@ -139,6 +139,7 @@
 .endm
 	HUFFMAN_DECODE_8 )" ARGS R"(
 .purgem HUFFMAN_DECODE_8
+	.att_syntax
 	)"
 		: "=a"(ret), "=D"(clobber), "=S"(clobber), "=b"(clobber)
 		: "D"(pDstBegin), "a"(pDstEnd), "S"(pSrcBegin), "b"(pDecodeTable),
--- utvideo-18.2.1/utv_core/HuffmanCode_x64_HuffmanEncode.cpp.orig	2017-07-16 08:59:00.000000000 +0200
+++ utvideo-18.2.1/utv_core/HuffmanCode_x64_HuffmanEncode.cpp	2017-07-22 20:38:58.624140074 +0200
@@ -98,6 +98,7 @@
 .endm
 	HUFFMAN_ENCODE )" ARGS R"(
 .purgem HUFFMAN_ENCODE
+	.att_syntax
 	)"
 		: "=a"(ret), "=D"(clobber), "=S"(clobber), "=d"(clobber)
 		: "D"(pDstBegin), "S"(pSrcBegin), "a"(pSrcEnd), "d"(pEncodeTable)
